datalad/__init__.py

```diff
@@ -118,12 +121,19 @@ def enable_librarymode():
     if cfg.getbool('datalad.runtime', 'librarymode', False) \
     else 'application'
 
+
 from datalad.utils import (
     get_encoding_info,
     get_envvars_info,
     getpwd,
+    is_interactive as ut_is_interactive,
 )
 
+# TODO: Interactivity auto-false when librarymode is enabled?
+is_interactive = ut_is_interactive()
+# Propagate to subprocesses:
+os.environ['DATALAD_UI_INTERACTIVE'] = 'true' if is_interactive else 'false'
+
```

Comment:
> but datalad code in a subprocess also runs this very code itself.

Yes, but suggestion is not entirely true. A python API user can set a config override. This would be lost on a subprocess. Very much like `enable_librarymode`.
Of course, such caller can set an env var themselves, but it requires awareness of possible datalad subprocesses (not only special remotes, but `sshrun`, git-credential-help, ..). I think, that's a needless burden.

Reply:
Actually, just realized, I didn't actually solve my aim here. Will not resolve yet.

Reply:
I made #7352 to address the general case instead.

Reply:
Removed propagation from here in favor of above mentioned issue.